tests/kuttl/tests/collapsed/01-assert-collapsed-cell.yaml

```diff
@@ -77,12 +77,10 @@ spec:
       # separation implemented.
       cellTemplates:
           cell0:
-            cellDatabaseUser: nova_cell0
             conductorServiceTemplate:
               replicas: 0
             hasAPIAccess: true
           cell1:
-            cellDatabaseUser: nova_cell1
             cellDatabaseInstance: openstack
```

Comment:
I think I see what is happening here. Originally we defined the cellDatabaseUser fields here explicitly as the defaulting only works properly if the celltemplate is not defined at all. If a field is defined in it then the rest is not defaulted according to the kubebuilder defaults. Which make sense as the defaults are defined in the template field level https://github.com/openstack-k8s-operators/nova-operator/blob/2fc1351673afc28eb493b8d4d9feb2810197a199/api/v1beta1/nova_types.go#L52 
not on the individual field level within the template struct. (we tried to do that defaulting as well but that fails due to webhhooks ser/des the struct). So the solution here is to explicitly confiuger the database account name as well. In the other sample files we don't define the cell template at all and there the defaulting works.

Reply:
actually what I wrote here belongs to the collapsed cell sample file that is used as an input in the kuttl test. This is just the assert where we could drop the user and don't have to assert the account. But the sample file needs to explicitlly provide the account due to defaulting quirks